client/my-sites/domains/controller.jsx

```diff
@@ -171,13 +175,12 @@ const googleAppsWithRegistration = ( context, next ) => {
 		page( '/checkout/' + siteSlug );
 	};
 
-	analytics.pageView.record(
-		'/domains/add/:site/google-apps',
-		'Domain Search > Domain Registration > Google Apps'
-	);
-
 	context.primary = (
 		<Main>
+			<PageViewTracker
+				path="/domains/add/:domain/google-apps/:site"
```

Comment:
Would suggest adding these to the `lib/paths` but even w/o it this is looking good.

Reply:
@aidvu I'm actually curious why folks like the `lib/paths` pattern. It makes searching for routes in our editors incredibly difficult.

Reply:
Reusability. :) We reuse these `paths` often in the `domain-management` code. I can't think of a better way to do it, and would rather not have strings hardcoded in every place we use them. I'm open to suggestions tho! 

What comes to mind is a list of constants, and a simple `replacer` function to which you pass an object with `pathVar: value`?

Reply:
👍 I don't have issues with reuse, but we can still write these in a way that makes it more searchable or at least readable without needing to jump through multiple functions to construct the path in our head.

For example:
```jsx
function editorPathFromSite( site ) {
	let path = '',
		siteSlug;

	if ( site ) {
		siteSlug = typeof site === 'object' ? site.slug : site;
		path = '/' + siteSlug;
	} else if ( site && typeof site === 'object' ) {
// Aside: reading this, when do we ever hit this case? An object should be truthy
		path = '/' + site.ID + '/new';
	}

	return path;
}

export function newPost( site ) {
	const sitePath = editorPathFromSite( site );
	return '/post' + sitePath;
}
```

Might also be written as:
```jsx
export function newPost( site ) {
    const siteSlug = get( site, [ 'slug' ], site );
    return siteSlug ? `/post/${ siteSlug }` : `/post`;
}
```

Reply:
Totally. What I had in mind would be something like:

`lib/paths`
```
export const gappsPath = '/domains/add/:domain/google-apps/:site';
```
`lib/paths/utils.js`
```
export function buildPath( path, pathVariables ) {
  _.forEach( pathVariables, function( value, key ) {
    path = path.replace( ':' + key, value );
  });

  return path;
}
```

Pseudo-code, but hope it can pass the idea. :)

Reply:
@aidvu Would suggest adding these to the `lib/paths` but even w/o it this is looking good.

If you don't mind, I'd rather stay out of `lib/paths` as that's not the only hardcoded path in the Domains section.

_Also I really don't like `lib/paths`_ 😄

Reply:
Totally don't mind. I approved it, and said that it'd be nice, but tbh, I like the above idea discussed with @gwwar, and might take on it myself later. 

This is a huge improvement over what it was, so :shipit:!